azurelinuxagent/ga/exthandlers.py

```diff
@@ -807,7 +832,18 @@ def handle_enable(self, ext_handler_i, extension):
         # We go through the entire process of downloading and initializing the extension if it's either a fresh
         # extension or if it's a retry of a previously failed upgrade.
         if current_handler_state == ExtHandlerState.NotInstalled or current_handler_state == ExtHandlerState.FailedUpgrade:
-            self.__setup_new_handler(ext_handler_i, extension)
+
+            # Check extension policy and raise error if disallowed (e.g., not in allowlist, unsigned when policy requires signature).
+            extension_is_signed = ext_handler_i.ext_handler.encoded_signature != ""
+            self._policy_engine.check_extension_policy(ext_handler_i.ext_handler.name, extension_is_signed)
+
+            # Determine whether to ignore signature validation errors based on policy and conf flag. Errors should be ignored
```

Comment:
I don't think policy should be involved in this decision

Reply:
That logic is only temporary, which should be made very clear in the comments and implementation. Probably wrapping this logic in a helper method would be a good idea, in case it needs to be used in more than one place.